src/domain/users/user-organizations.repository.ts

```diff
@@ -117,51 +120,44 @@ export class UsersOrganizationsRepository
 
     const invitedAddresses = args.users.map((user) => user.address);
     const invitedWallets = await this.walletsRepository.find({
-      where: {
-        address: In(invitedAddresses),
-      },
-      relations: {
-        user: true,
-      },
+      where: { address: In(invitedAddresses) },
+      relations: { user: true },
     });
-
     const invitations: Array<Invitation> = [];
 
     await this.postgresDatabaseService.transaction(async (entityManager) => {
       for (const userToInvite of args.users) {
-        // Find existing User via Wallet
-        const invitedWallet = invitedWallets.find((wallet) => {
+        // Find existing User via Wallet or create new User and Wallet.
+        const wallet = invitedWallets.find((wallet) => {
           return isAddressEqual(wallet.address, userToInvite.address);
         });
-        let invitedUserId = invitedWallet?.user.id;
-
-        // Otherwise create User and Wallet
-        if (!invitedUserId) {
-          invitedUserId = await this.usersRepository.create(
-            'PENDING',
-            entityManager,
-          );
-
-          await this.walletsRepository.create(
-            {
-              walletAddress: userToInvite.address,
-              userId: invitedUserId,
-            },
-            entityManager,
-          );
+        const userIdToInvite = wallet
+          ? wallet.user.id
+          : await this.createUserAndWallet({
+              entityManager,
+              address: userToInvite.address,
+            });
+
+        try {
+          await entityManager.insert(DbUserOrganization, {
+            user: { id: userIdToInvite },
+            organization: org,
+            name: userToInvite.name,
+            role: userToInvite.role,
+            status: 'INVITED',
+            invitedBy: adminAddress,
+          });
+        } catch (err) {
+          if (isUniqueConstraintError(err)) {
+            throw new UniqueConstraintError(
+              `${userToInvite.address} is already in this space or has a pending invite.`,
+            );
+          }
+          throw err;
```

Comment:
We should prioritise a filter for these error types, but in a follow up.

Reply:
Agreed, will note this!